### PR TITLE
Improvements for 00002_log_and_exception_messages_formatting

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

A few commits to improve `00002_log_and_exception_messages_formatting`. Best reviewed per commit.

The last one is super big as it changes how ErrnoExceptions are thrown so we have a similar format string as we do with DB::Exception (I hope). cc @tavplubix since you added both the check and the previous improvements to DB::Exception.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit b269f87f4c76671d34aec2e1dc7489b1b53281d1 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57910/b269f87f4c76671d34aec2e1dc7489b1b53281d1/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7225728986">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57910/b269f87f4c76671d34aec2e1dc7489b1b53281d1/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Common/AsynchronousMetrics.cpp [hunk 1]
```diff
@@ -797,7 +797,7 @@ void AsynchronousMetrics::update(TimePoint update_time)
 
             int64_t hz = sysconf(_SC_CLK_TCK);
             if (-1 == hz)
-                throwFromErrno("Cannot call 'sysconf' to obtain system HZ", ErrorCodes::CANNOT_SYSCONF);
```
Comment (alexey-milovidov) [inline src/Common/AsynchronousMetrics.cpp:800]: But then you lose errno. We don't want to lose this detail.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: src/Common/AsynchronousMetrics.cpp (hunk 1 above)
Comment (Algunenano) [reply src/Common/AsynchronousMetrics.cpp:800]: It's still being captured

https://github.com/ClickHouse/ClickHouse/pull/57910/files#diff-9884cbf5c6e44559e86a36073ca63b3dd91929be7123373310d6e824af4504a6R190-R193

Comment (tavplubix) [reply src/Common/AsynchronousMetrics.cpp:800]: Why do we still need `ErrnoException::throwWithErrno` then?

**Review by alexey-milovidov — COMMENTED** [review]
Thank you!

File: src/Common/AsynchronousMetrics.cpp (hunk 1 above)
Comment (Algunenano) [reply src/Common/AsynchronousMetrics.cpp:800]: For the case where errno is saved, other ops are done and then the exception is raised. Also when the error code doesn’t come from errno but from the result of an operation. I think there is only a handful of calls

**Review by alexey-milovidov — APPROVED** [review]

**Comment by Algunenano** [comment]
I wonder why `ClickHouse special build check` is not in the CI reporting that the FreeBSD and clang-tidy builds had failed

**Pull request merged**